### DISCUSSION TITLE
How to inject Tracer using .WithTracing?

### DISCUSSION THREAD

**Discussion by jade-lucas** [open-telemetry/opentelemetry-dotnet#4646] [General]
This documentation shows how to inject a Tracer. https://opentelemetry.io/docs/instrumentation/net/shim/#injecting-a-tracer-with-aspnet-core

Is there away to inject it via .WithTracing? I didn't see any examples using this approach nor any anything on the tracing builder to do so. I could be just not seeing it or do I need to break the fluent calls and do it how it show in the previous example?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started-aspnetcore/README.md
```
builder.Services.AddOpenTelemetry()
    .ConfigureResource(resource => resource
        .AddService(serviceName: builder.Environment.ApplicationName))
    .WithTracing(tracing => tracing
        .AddAspNetCoreInstrumentation()
        .AddConsoleExporter());
```
I am trying to achieve some custom tracing for things that are not implemented in OTEL.

**Comment by vishweshbankwar** [comment]
@jade-lucas If you are looking to create custom spans you can do so via ActivitySource. https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Api#instrumenting-a-libraryapplication-with-net-activity-api

Take a look at this example as well https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Api#instrumenting-a-libraryapplication-with-net-activity-api
